### DISCUSSION TITLE
Malt Soft Shadows

### DISCUSSION THREAD

**Discussion by Psi25** [bnpr/Malt#440] [Q&A] [answered]
Is it possible to get a general soft shadow in Malt? I've wondered if it's possible to make all shadows look soft or make shadows soft based with falloff based on light distance.

**Comment by pragma37** [comment]
There's nothing built-in, although implementing EEVEE-style soft shadows should be easy since it only requires jittering the light position/direction each sample.

Falloff based on light distance is what already happens by default with point and spot lights.

**Comment by aVersionOfReality** [comment, marked as answer]
If you wan to replicate any of Eevee's light behavior, most of it is in: source\blender\draw\engines\eevee\shaders\lights_lib.glsl

I've recently had to replicate its shading and light falloff (but not shadows) so feel free to ask me if you get stuck.
